ligo/main.mligo

```diff
@@ -207,21 +225,118 @@ let set_position (s : storage) (i_l : tick_index) (i_u : tick_index) (i_l_l : ti
       }
     )
 
-let get_time_weighted_sum (s : storage) (c : views contract) : result =
-    ([Tezos.transaction (IC_sum s.time_weighted_ic_sum) 0mutez c], s)
+// Allocate slots in a buffer in [start, to_stop) indices range.
+// These slots will be filled with dummy values.
+let rec allocate_buffer_slots (buffer, start, to_stop : timed_cumulatives_buffer * nat * nat) : timed_cumulatives_buffer =
+    if start >= to_stop
+    then buffer
+    else
+        let new_buffer = {buffer with map = Big_map.add start dummy_timed_cumulatives buffer.map}
+        in allocate_buffer_slots(new_buffer, start + 1n, to_stop)
+
+// Increase the number of stored accumulators.
+let increase_observation_count (s, new_val : storage * nat) : result =
+    let buffer = s.cumulatives_buffer in
+    if buffer.reserved_length >= new_val
+    then (([] : operation list), s)
+    else
+        let new_buffer = allocate_buffer_slots(buffer, buffer.last + 1n, buffer.first + buffer.reserved_length)
```

Comment:
Perhaps I'm missing something, but isn't an update to the `buffer.reserved_length` to be done here?

Reply:
Right :facepalm: 

It's sad to write code without tests :/